test/unit/BFactory.t.sol

```diff
@@ -1,251 +1,142 @@
+    IBPool pool = factory.newBPool();
 
-// solhint-disable-next-line no-empty-blocks
-contract BFactory_Unit_Constructor is BFactoryTest, BaseBFactory_Unit_Constructor {}
-
-contract BFactory_Unit_IsBPool is BFactoryTest {
-  /**
-   * @notice Test that a valid pool is present on the mapping
-   */
-  function test_Returns_IsValidPool(address _pool) public {
-    // Writing TRUE (1) to the mapping with the `_pool` key
-    vm.store(address(bFactory), keccak256(abi.encode(_pool, uint256(0))), bytes32(uint256(1)));
-    assertTrue(bFactory.isBPool(address(_pool)));
+    // it should add the newBPool to the list of pools
+    assertTrue(factory.isBPool(address(_newBPool)));
+    // it should return the address of the new BPool
+    assertEq(address(pool), _newBPool);
   }
 
-  /**
-   * @notice Test that a invalid pool is not present on the mapping
-   */
-  function test_Returns_IsInvalidPool(address _randomPool) public view {
-    vm.assume(_randomPool != address(0));
-    assertFalse(bFactory.isBPool(_randomPool));
-  }
-}
-
-abstract contract BaseBFactory_Unit_NewBPool is Base {
-  /**
-   * @notice Test that the pool is set on the mapping
-   */
-  function test_Set_Pool() public {
-    IBPool _pool = bFactory.newBPool();
-    assertTrue(bFactory.isBPool(address(_pool)));
-  }
-
-  /**
-   * @notice Test that event is emitted
-   */
-  function test_Emit_Log(address _randomCaller) public {
-    assumeNotForgeAddress(_randomCaller);
-
-    vm.expectEmit();
-    address _expectedPoolAddress = vm.computeCreateAddress(address(bFactory), 1);
-    emit IBFactory.LOG_NEW_POOL(_randomCaller, _expectedPoolAddress);
-    vm.prank(_randomCaller);
-    bFactory.newBPool();
+  function test__newBPoolWhenCalled() external {
+    address _futurePool = vm.computeCreateAddress(address(factory), 1);
+    address _newBPool = address(factory.call__newBPool());
+    assertEq(_newBPool, _futurePool);
+    // it should deploy a new BPool
+    assertEq(_newBPool.code, address(new BPool()).code);
   }
 
-  /**
-   * @notice Test that msg.sender is set as the controller
-   */
-  function test_Set_Controller(address _randomCaller) public {
-    assumeNotForgeAddress(_randomCaller);
+  function test_SetBLabsRevertWhen_TheSenderIsNotTheCurrentSetBLabs(address _caller) external {
+    // Pre-condition
+    vm.assume(_caller != factoryDeployer);
 
-    vm.prank(_randomCaller);
-    IBPool _pool = bFactory.newBPool();
-    assertEq(_randomCaller, _pool.getController());
-  }
+    // Post-condition
+    // it should revert
+    vm.expectRevert(IBFactory.BFactory_NotBLabs.selector);
 
-  /**
-   * @notice Test that the pool address is returned
-   */
-  function test_Returns_Pool() public {
-    address _expectedPoolAddress = vm.computeCreateAddress(address(bFactory), 1);
-    IBPool _pool = bFactory.newBPool();
-    assertEq(_expectedPoolAddress, address(_pool));
+    // Action
+    vm.prank(_caller);
+    factory.setBLabs(makeAddr('newBLabs'));
   }
```

Comment:
starting to like this, very easy to read 👌

Reply:
i'd remove the banners `// Action` and `// Post-condition`, seems like extra work and doesn't apply to all of the cases